superset/assets/src/visualizations/nvd3/NVD3Vis.js

```diff
@@ -355,12 +356,31 @@ function nvd3Vis(element, props) {
             }
             const xField = formData.granularitySqla;
             const yField = findYAxisField(yColumn, publishedColumns);
-
+            let xFieldVal;
+            let yFieldVal
+            let xValueChanged;
+            let yValueChanged;
             if (xField != undefined && e.point) {
-              const xFieldVal = getXAxisFieldVal(xField, e.point.x, columns);
-              onAddFilter(xField, xFieldVal, false, !yField);
+              xFieldVal = getXAxisFieldVal(xField, e.point.x, columns);
+              xValueChanged = selections ? selections['point'].x != e.point.x  : true;
+            }
+
+            if (yField != undefined && e.point) {
+              yFieldVal = e.point.y;
+              yValueChanged = selections ? selections['point'].y != yFieldVal  : true;
             }
-            if (yField != undefined && e.point) onAddFilter(yField, e.point.y, false, true);
+
+            if ( (xValueChanged && yValueChanged ) ||  (yValueChanged && !xValueChanged)) {
+              onAddFilter(xField, xFieldVal, false, false);
+              onAddFilter(yField, yFieldVal, false, true);
+            }
+            else if (xValueChanged && !yValueChanged) {
+              onAddFilter(yField, yFieldVal, false, false);
+              onAddFilter(xField, xFieldVal, false, true);
+            }
+
+            // set selections
+            selections = {'point': e.point}
```

Comment:
either make varibale as `selection` or put in `[]`.  I like the later as it will allow multiple selection scope in future.
in code for now we can say `selections[0]`

Reply:
did it selection right now